client/my-sites/checkout/composite-checkout/hooks/use-prepare-products-for-cart.ts

```diff
@@ -533,13 +551,17 @@ function createItemToAddToCart( {
 		createRequestCartProduct( {
 			product_id: productId,
 			product_slug: productSlug,
-		} )
+		} ),
+		{ isJetpackUserlessCheckout }
```

Comment:
I don't fully grok why we need this `options` object, especially as an unknown `Record` type. Presumably (since `addContextToProduct` returns a `RequestCartProduct`), any properties added to the product's `extra` must already be supported by `RequestCartProductExtra`, so we could do something like the following (or other variations like having a dedicated function to add this property):

```js
createRequestCartProduct({
  product_id: productId,
  product_slug: productSlug,
  ...( isJetpackUserlessCheckout ? { extra: { isJetpackUserlessCheckout: true } } : {} )
})
```

Just generally, it also seems like `addContextToProduct` should just add `context`.